tests/utils.py

```diff
@@ -75,6 +70,30 @@ def override_env(env):
         os.environ.update(original)
 
 
+@contextlib.contextmanager
+def override_env(env):
+    """
+    Temporarily override ``os.environ`` with provided values::
+
+        >>> with self.override_env(dict(DD_TRACE_DEBUG=True)):
+            # Your test
+    """
+    return _override_env(env)
+
+
+def override_decorated_env(env):
+    """Decorator for test functions that overrides environment variables passed as a dictionary"""
+
+    def _decorator(fn):
+        def wrapper(fn, *args, **kwargs):
+            with override_env(env):
+                return fn(*args, **kwargs)
+
+        return decorator.decorator(wrapper, fn)
+
+    return _decorator
```

Comment:
Putting a decorator on the test functions leads to fewer lines of code changed than indenting all of the test function bodies to be under that context manager. I'm actually going to do that, though, because the current implementation requires regenerating every requirement lockfile, which is definitely worse.